homeworks/25/instagram-api/src/main/java/org/elsys_bg/servlets_hw/Posts.java

```diff
@@ -0,0 +1,304 @@
+package org.elsys_bg.servlets_hw;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Types;
+import java.io.PrintWriter;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import javax.naming.InitialContext;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.sql.DataSource;
+
+import org.json.JSONObject;
+
+
+@WebServlet("/posts/*")
+public class Posts extends HttpServlet {
+
+	private Connection connection;
+
+	@Override
+	public void init() throws ServletException {
+		try {
+			DataSource ds = (DataSource) new InitialContext().lookup("java:comp/env/jdbc/instagram_api");
+			connection = ds.getConnection();
+
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+	}
+	
+	@Override
+	protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {     
+        PrintWriter pw = resp.getWriter();
```

Comment:
The code will look better if you add "tap" before writing inside the function body.